File: React Native/RN+OC/iOS/RN+OC/ViewController.m
//
//  ViewController.m
//  RN+OC
//
//  Created by Yunan Xu on 17/12/2017.
//  Copyright © 2017 xuyunan0113@gmail.com. All rights reserved.
//

#import "ViewController.h"
#import <React/RCTRootView.h>
#import "RNViewController.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad
{
    [super viewDidLoad];
    self.navigationController.view.backgroundColor = [UIColor whiteColor];
}

- (IBAction)gotoRNPage:(id)sender
{
    NSString *version = [[[NSBundle mainBundle] infoDictionary] objectForKey:@"CFBundleShortVersionString"];
    
    NSURL *jsCodeLocation = [NSURL URLWithString:@"http://localhost:8081/pages/about.bundle?platform=ios"];
    RCTRootView *rootView = [[RCTRootView alloc] initWithBundleURL: jsCodeLocation
                                                        moduleName: @"About"
                                                 initialProperties: @{@"version": version}
                                                     launchOptions: nil];
    UIViewController *vc = [[UIViewController alloc] init];
    vc.title = @"关于";
    vc.view = rootView;
    
    [self.navigationController pushViewController:vc animated:YES];
}

- (IBAction)gotoCommunicationPage:(id)sender
{
    RNViewController *vc = [[RNViewController alloc] init];
    vc.title = @"RN调用OC示例";
    [self.navigationController pushViewController:vc animated:YES];
}

- (void)didReceiveMemoryWarning
{
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}


@end

File: iOS/CoreText1/CoreText1/ViewController.swift
//
//  ViewController.swift
//  CoreText1
//
//  Created by Yunan Xu on 03/12/2017.
//  Copyright © 2017 xuyunan0113@gmail.com. All rights reserved.
//

// 示例内容:
// 1. CoreText基础
// 2. 读取文件文本内容, 使用CoreText绘制, 控制富文本属性, 设置样式...
// 3. 使用空白字符站位, 预留空间, 绘制图片, 实现图文混排

import UIKit

class ViewController: UIViewController {

    @IBOutlet weak var ctview: CTView!
    
    override func viewDidLoad() {
        super.viewDidLoad()
        guard let file = Bundle.main.path(forResource: "kenan", ofType: "txt") else { return }
        do {
            let text = try String(contentsOfFile: file, encoding: .utf8)
            ctview.str = text
        } catch _ {
            
        }
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }


}

